@ecomplus/storefront-components/src/js/ProductVariations.js

```diff
@@ -131,5 +142,23 @@ export default {
       deep: true,
       immediate: true
     }
+  },
+
+  mounted () {
+    if (this.variationFromUrl && Array.isArray(this.product.variations)) {
+      const selectedVariation = this.product.variations.find(variation => variation._id === this.variationFromUrl)
+      if (selectedVariation) {
+        const { specifications } = selectedVariation
+        for (const spec in specifications) {
+          if (Object.hasOwnProperty.call(specifications, spec)) {
+            const specObject = specifications[spec][0]
+            const index = this.variationsGrids[spec].findIndex(options => options === specObject.text)
+            this.$nextTick(() => {
+              this.selectOption(specObject.text, spec, index)
+            })
+          }
+        }
```

Comment:
```suggestion
        const secs = Object.keys(specifications)
        const nextSpec = (specIndex = 0) => {
          const spec = specs[specIndex]
          if (specs[specIndex] && specifications[spec] && specifications[spec].length === 1) {
            const specText  = specifications[spec][0].text
            if (this.variationsGrids[spec].find(option => option === specText)) {
              this.$nextTick(() => {
                this.selectOption(specObject.text, spec, this.orderedGrids.indexOf(spec))
                nextSpec(specIndex + 1)
              })
            }
          }
        }
        nextSpec()
```

Reply:
Acho que tinha bastante coisa errada aí na verdade:
1. Não trata variações com duas cores (ou dois valores de uma grade qualquer), continuei não tratando mas pelo menos parei de selecionar a primeira opção (que pode ser errada) nesse caso, não seleciona nada..
2. Eu acho que não funcionaria com mais de uma grade porque você adiciona "ao mesmo tempo" vários callbacks pro próximo ciclo (`$nextTick`), na verdade teria que ser um select a cada ciclo;
3. Os argumentos de `this.selectOption` estavam errados, o terceiro argumento não é o index da opção no `variationGrids[index]`, é o index do grid https://github.com/ecomplus/storefront/blob/16e5aa5ee7b07944e5256b6e8f580ed982d55e8b/%40ecomplus/storefront-components/src/html/ProductVariations.html#L6